cluster/manager/manager.go

```diff
@@ -689,15 +693,19 @@ func (c *ClusterManager) joinCluster(
 	// there may be a mismatch between db updates from listeners and
 	// cluster db state.
 
-	kvdb := kvdb.Instance()
-	kvlock, err := kvdb.LockWithID(clusterLockKey, c.config.NodeId)
+	lockKey := path.Join(c.config.NodeId, self.DataIp, fn)
+	lockTryDuration := defaultClusterLockTryDuration
+	if c.config.SnapLockTryDurationInMinutes > 0 {
+		lockTryDuration = time.Duration(c.config.SnapLockTryDurationInMinutes) * time.Minute
+	}
+	kvlock, err := kvdb.Instance().LockWithTimeout(clusterLockKey, lockKey, lockTryDuration, kvdb.DefaultLockTryDuration)
```

Comment:
Last parameter is called "lock hold duration" in kvdb.go. Just making sure that we are passing the right value here.

Names of these parameters are too similar to each other. Wonder if there is some way to clarify what's what using more specific names. Just a nit.